### PR TITLE
Feat array codeblock

### PR DESCRIPTION
**Pull request by Fabianstw**
This PR includes an easy way to display arrays consistently across the website.

Currently there is nothing to display, because there are no current example 😅 

### Also included:
- bun update 
- shadcn/ui separator (currently unused)
- removed assertion inside BasicMultipleChoice, because bun run test failed

### CODE REVIEW

File: shared/src/api/BasicMultipleChoiceQuestion.ts [hunk 1]
```diff
@@ -121,8 +121,7 @@ export function basicMultipleChoiceMetaGenerator(
 
     random.shuffle(answers)
 
-    const markdown =
-      questions[i].frame === undefined ? ownt["text"] : questions[i].frame!(l, ownt["text"])
+    const markdown = questions[i].frame!(l, ownt["text"])
```
Comment (AbcAeffchen) [inline shared/src/api/BasicMultipleChoiceQuestion.ts:124]: Is change breaks the functionality.

The linter complains about the `!` at the end. This is an implicit assertion that `.frame` is not undefined there. But we already checked for that case, which makes the assertion unnecessary. 

So the correct change would be to just remove `!` resulting in 
```
questions[i].frame === undefined ? ownt["text"] : questions[i].frame(l, ownt["text"])
```

**Comment by AbcAeffchen** [comment]
Is the a specific reason for updating the packages? Any bug-fixes or new features we need?
I'm confused that the update also reflects in package.json. Shouldn't an update only change the lock file?

File: shared/src/api/BasicMultipleChoiceQuestion.ts (hunk 1 above)
Comment (holgerdell) [reply shared/src/api/BasicMultipleChoiceQuestion.ts:124]: The ! is necessary in typescript < 5.5, only >= 5.5 is able to infer this correctly.

**Comment by holgerdell** [comment]
Please don't run `bun update` at the moment. I checked and typescript 5.5 brings cool new features that are not reflected in eslint yet. We are blocked from upgrading until eslint `v8` becomes stable and typescript-eslint upgrades its rules to typescript 5.5.

**Comment by holgerdell** [comment]
> Is the a specific reason for updating the packages? Any bug-fixes or new features we need? I'm confused that the update also reflects in package.json. Shouldn't an update only change the lock file?

No, package.json specifies minimum required versions or exact versions according to some rules, while the lock-file always specifies an exact version that is known to work.

File: shared/src/api/BasicMultipleChoiceQuestion.ts (hunk 1 above)
Comment (AbcAeffchen) [reply shared/src/api/BasicMultipleChoiceQuestion.ts]: Ah, this explains it. The PR includes an update from typescript 5.4 to 5.5

Comment (Fabianstw) [reply shared/src/api/BasicMultipleChoiceQuestion.ts]: I changed it back and will add the ! if I go back to < 5.5

**Comment by AbcAeffchen** [comment]
> > Is the a specific reason for updating the packages? Any bug-fixes or new features we need? I'm confused that the update also reflects in package.json. Shouldn't an update only change the lock file?
> 
> No, package.json specifies minimum required versions or exact versions according to some rules, while the lock-file always specifies an exact version that is known to work.

Yes. Hence my confusion.

**Comment by Fabianstw** [comment]
> Please don't run `bun update` at the moment. I checked and typescript 5.5 brings cool new features that are not reflected in eslint yet. We are blocked from upgrading until eslint `v8` becomes stable and typescript-eslint upgrades its rules to typescript 5.5.

Oh okay, I did not knew that. I know how to rollback the package.json file, but how can I get an old version of the bun.lockb?

**Comment by Fabianstw** [comment]
Or is it enough to write `"typescript": "^5.4.5",` into the package.json file? 
Are the other changes okay?

**Comment by AbcAeffchen** [comment]
I don't know what the other updates change. This is why I asked for the reasons of the update. Such changes can bring all sorts of unexpected issues and should only be done if necessary, e.g. a feature we want, of for bug-fixes.

**Comment by AbcAeffchen** [comment]
You should be able to get an older version of both files from the git history.

**Comment by Fabianstw** [comment]
Okay. Ill revert to older versions

**Comment by Fabianstw** [comment]
It should all be back now 👍 
 
bun.lockb still has changes (I think because of `"@radix-ui/react-separator": "^1.1.0"`

**Comment by AbcAeffchen** [comment]
What is the difference between react-separator and plain HTML `<hr>`?

**Comment by Fabianstw** [comment]
> What is the difference between react-separator and plain HTML `<hr>`? 

I think not much. But you can easily access it and we'll be consistent across the webpage. 

But because it's currently unused, I can remove it again and we will add it later if we need it

**Comment by AbcAeffchen** [comment]
I don't know how you plan to use it. If you have a use case that is not covered by `<hr>` than it is fine. Otherwise we should just use `<hr>` and save use the additional package. This would still be consistent. I mean, are using default HTML all over the place.

**Comment by AbcAeffchen** [comment]
Ok, then is my final comment, that package.json and the lockfile should just be rolled back completely. (package.json has only a empty line removed, the lockfile has some unknown changes)

**Pull request merged**